lib/install.js

```diff
@@ -438,18 +477,32 @@ function download (gyp, env, url) {
     requestOpts.ca = readCAFile(cafile)
   }
 
-  // basic support for a proxy server
-  var proxyUrl = gyp.opts.proxy
-              || env.http_proxy
-              || env.HTTP_PROXY
-              || env.npm_config_proxy
-  if (proxyUrl) {
-    if (/^https?:\/\//i.test(proxyUrl)) {
-      log.verbose('download', 'using proxy url: "%s"', proxyUrl)
-      requestOpts.proxy = proxyUrl
-    } else {
-      log.warn('download', 'ignoring invalid "proxy" config setting: "%s"', proxyUrl)
+  var noProxyUrl = gyp.opts.no_proxy
+    || env.no_proxy
+    || env.NO_PROXY
+    || env.npm_config_no_proxy
+    || ''
+
+  if (noProxyUrl === '*' || (noProxyUrl !== '' && uriInNoProxy(url, noProxyUrl))) {
```

Comment:
Long line; can you keep it under 80 columns?  The parens around the `&&` aren't necessary.